sql/parse/parse.go

```diff
@@ -165,7 +165,25 @@ func ParseColumnTypeString(ctx *sql.Context, columnType string) (sql.Type, error
 
 func convert(ctx *sql.Context, stmt sqlparser.Statement, query string) (sql.Node, error) {
 	if ss, ok := stmt.(sqlparser.SelectStatement); ok {
-		return convertSelectStatement(ctx, ss)
+		node, err := convertSelectStatement(ctx, ss)
+		if err != nil {
+			return nil, err
+		}
+		// *Into should be outer layer around select/union statements.
+		var into *sqlparser.Into
+		switch s := ss.(type) {
```

Comment:
Do we need to do anything for `ParenSelect`? Is't the only other current implementation of `SelectStatement` and isn't covered in this switch. I'm wondering if this logic can be handled more generically, without having to switch on type. That would keep the code a little leaner/cleaner and also prevent potential bugs in the future if new implementations of SelectStatement are added, but someone doesn't know to update this code. I haven't dug into the details to see if that could really work, just a question, so let me know if I'm missing something.

Reply:
Having into defined in ParenSelect is unlikely since any select statement in parentheses is caught in parser to not to have into defined, but you're right, it's more clear and more clean to have it as a case than ignore it.

Reply:
Nice. Yeah, I think removing the switch on type and just working with SelectStatement, instead of it's subtypes, will make the code a little bit simpler/smaller and will also automatically keep this code working if we add a new type of SelectStatement in the future that needs to support `into` behavior.